JarrydWorland/2021_Capstone_GP2_Team
# Refactor weapons

## Changes include
- Added TagBehaviour (711ad1b)
- Added dictionary extension methods (c800980)
- Changed item behaviour start method to public (048f073)
- Added projectiles (34930e1)
- Added weapons (6508762)
- Added health (ca14246)

## Issues
- The use item animation does not properly stop when changing the equipped weapon to another. This is a limitation of the item system at the moment and should be fixed in a separate PR.
- The sprites used for the projectiles and weapons are placeholder.
- For now projectiles will go through walls as wall colliders on rooms have not been re-added.

## How to test
Place an Inventory object on the camera, and place a Player and DefaultWeaponUpgraded on the floor, ensure you can pick it up, equip it, fire it, and drop it correctly.
